### DISCUSSION TITLE
Support Apache Hudi

### DISCUSSION THREAD

**Discussion by shrkant** [apache/amoro#2258] [General]
Hi Team,

I am a software engineer working on a large-scale data product in the ETL field. Recently, we have been exploring the control plane for our lakehouse tables, and we noticed that it shares similar objectives with Amoro.

We have explored Amoro and gained a basic understanding of its architecture and various components/services. I would like to know if there are any plans to include support for Apache Hudi format in future releases.

Thank you.

Best regards,
Shrikant Bang.

**Comment by shidayang** [comment]
> Hi Team,
> 
> I am a software engineer working on a large-scale data product in the ETL field. Recently, we have been exploring the control plane for our lakehouse tables, and we noticed that it shares similar objectives with Amoro.
> 
> We have explored Amoro and gained a basic understanding of its architecture and various components/services. I would like to know if there are any plans to include support for Apache Hudi format in future releases.
> 
> Thank you.
> 
> Best regards, Shrikant Bang.

Thanks for your bring this discussion up. We are very willing to integrate Hudi, and if any community member is willing to cover this work, Amoro is happy to put the integration of Hudi on the agenda immediately.

**Comment by zhoujinsong** [comment]
@shrkant Hi, I'm glad to see you bringing up this topic at this time.

One important feature of Amoro is its support for multiple data lake table formats. The Amoro community has also included #220 in #1361, although it may not necessarily be completed in 2023. Nonetheless, this indicates that the community is committed to integrating support for the Apache Hudi format.

In version 0.6.0, Amoro not only completed the initial integration with Apache Paimon (Incubating) but also carried out some refactoring within the project to make it easier to integrate with other table formats in the future. 

So, I believe this is a good opportunity to start integrating Apache Hudi. Currently, what we may lack is more input on use cases from Hudi users. I warmly welcome you to share your expectations regarding Amoro's integration with Hudi. Of course, it would be even better if we could work together to drive the completion of this task.

**Comment by czy006** [comment]
The community has always had relevant plans for this part of the work. Woody’s support is of great significance to the improvement and supplement of Amoro’s ecology. Contributions and relevant suggestions are welcome.

**Comment by shrkant** [comment]
Hi @shidayang @zhoujinsong @czy006 

Thank you for responding! We are currently in the initial phases of evaluating and finalizing the objectives for the control plane system for Apache Hudi. Can I please ask to keep this thread open for some time?

Thank you.

Best regards,
Shrikant Bang.

**Reply by shidayang** [reply]
That's ok

**Comment by lalitbyteiq** [comment]
is there any update on apache Hudi?

**Reply by xxubai** [reply]
We support displaying Hudi's metadata, you can try. https://github.com/apache/amoro/pull/2877.
